File: lib/folavirt/remote/Pool.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-

from folavirt.remote.RemoteObject import RemoteObject
from folavirt.networking.Query import Query

class Pool(RemoteObject):
    def __init__(self, name = None, tcp = None):
        self.name = name

        if tcp != None:
            self.tcp = tcp   
        
    def getIscsiAddress(self):
        """
        Zwraca adres
        """
        # Tworzenie zapytania
        q = Query('getpooladdress', self.name)
        
        # Wykonywanie zapytania
        r = self.execute(q)
        
        return r.getData()
        
    def getVolumes(self):
        """
        Zwraca listę pul
        """
        # Zapytanie
        q = Query('getpoolvolumes', self.name)
        r = self.execute(q)
        
        return r.getData()
        
    def getType(self):
        """
        Zwraca typ
        """
        splitted = self.name.split("-")
        
        # Pula dynamiczna
        if splitted[0] == "Folavirt":
            if splitted[2] == "base":
                return "dynamic-base"
            if splitted[2] == "snapshot":
                return "dynamic-snapshot"
            
            return "dynamic"
        else:
            return "static"
        
    def getName(self):
        """
        Zwraca nazwę puli
        """
        return self.name

File: www/module/Application/src/Application/Model/Folavirt/Remote/Pool.php
<?php

namespace Application\Model\Folavirt\Remote;

use Application\Model\Folavirt\Networking\Query;

/**
 * Pula dyskowa
 * 
 * @package Application\Model\Folavirt\Remote
 */
class Pool
{
	public $agent;
	public $name;
	
	public function __construct($agent, $name)
	{
		$this -> agent = $agent;
		$this -> name = $name;
	}
	
	public function getName()
	{
		return $this -> name;
	}
	
	public function getAgent()
	{
		return $this -> agent;
	}
	
	public function getVolumes()
	{
		// Tworzenie zapytania
		$q = new Query();
		$q -> setCommand('getpoolvolumes');
		$q -> setData($this -> name);
		
		$r = $this -> getAgent() -> execute($q);
		
		return $r -> getData();
	}
	
	public function __toString()
	{
		return $this -> name;
	}
}
